File: miniprogram/src/types/api/check-in.ts
/**
 * 入住申请类型定义
 * @module types/api/check-in
 */

import type { IBaseEntity, IPageParams, IPageResult } from './common';

/**
 * 入住类型枚举
 */
export enum CheckInType {
  /** 长期住宿 */
  LONG_TERM = 1,
  /** 临时住宿 */
  TEMPORARY = 2,
}

/**
 * 入住申请信息
 */
export interface ICheckIn extends IBaseEntity {
  /** 学生ID */
  studentId: number;
  /** 学生姓名（冗余） */
  studentName?: string;
  /** 学号（冗余） */
  studentNo?: string;
  /** 入住类型 */
  checkInType: CheckInType;
  /** 校区编码 */
  campusCode?: string;
  /** 楼层编码 */
  floorCode?: string;
  /** 房间ID */
  roomId?: number;
  /** 房间编码 */
  roomCode?: string;
  /** 床位ID */
  bedId?: number;
  /** 床位编码 */
  bedCode?: string;
  /** 申请日期 */
  applyDate: string;
  /** 入住日期 */
  checkInDate?: string;
  /** 预计退宿日期（临时住宿） */
  expectedCheckOutDate?: string;
  /** 状态 */
  status: number;
  /** 审核人ID */
  approverId?: number;
  /** 审核人姓名 */
  approverName?: string;
  /** 审核时间 */
  approveTime?: string;
  /** 审核意见 */
  approveOpinion?: string;
  /** 申请原因/备注 */
  applyReason?: string;
  /** 备注 */
  remark?: string;
}

/**
 * 入住申请查询参数
 */
export interface ICheckInQueryParams extends IPageParams {
  /** 学生ID */
  studentId?: number;
  /** 学号 */
  studentNo?: string;
  /** 学生姓名 */
  studentName?: string;
  /** 入住类型 */
  checkInType?: CheckInType;
  /** 校区编码 */
  campusCode?: string;
  /** 楼层编码 */
  floorCode?: string;
  /** 状态 */
  status?: number;
  /** 申请日期开始 */
  applyDateStart?: string;
  /** 申请日期结束 */
  applyDateEnd?: string;
}

/**
 * 入住申请提交参数
 */
export interface ICheckInSubmitParams {
  /** 入住类型 */
  checkInType: CheckInType;
  /** 意向校区编码 */
  campusCode?: string;
  /** 意向楼层编码 */
  floorCode?: string;
  /** 意向房间ID */
  roomId?: number;
  /** 意向床位ID */
  bedId?: number;
  /** 入住日期 */
  checkInDate: string;
  /** 预计退宿日期（临时住宿） */
  expectedCheckOutDate?: string;
  /** 申请原因 */
  applyReason?: string;
}

/**
 * 入住申请审批参数
 */
export interface ICheckInApprovalParams {
  /** 申请ID */
  id: number;
  /** 审批结果 2=通过 3=拒绝 */
  status: 2 | 3;
  /** 分配的校区编码 */
  campusCode?: string;
  /** 分配的楼层编码 */
  floorCode?: string;
  /** 分配的房间ID */
  roomId?: number;
  /** 分配的床位ID */
  bedId?: number;
  /** 审批意见 */
  approveOpinion?: string;
}

/**
 * 入住申请分页响应
 */
export type ICheckInPageResult = IPageResult<ICheckIn>;

File: Application/src/main/java/com/project/backend/accommodation/entity/CheckIn.java
package com.project.backend.accommodation.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.project.core.entity.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 入住管理实体
 * 
 * @author 陈鸿昇
 * @since 2026-01-06
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_check_in")
@Schema(description = "入住管理实体")
public class CheckIn extends BaseEntity {

    @Schema(description = "学生ID")
    @TableField("student_id")
    private Long studentId;

    @Schema(description = "学生姓名（冗余）")
    @TableField("student_name")
    private String studentName;

    @Schema(description = "学号（冗余）")
    @TableField("student_no")
    private String studentNo;

    @Schema(description = "入住类型：1长期住宿 2临时住宿")
    @TableField("check_in_type")
    private Integer checkInType;

    @Schema(description = "校区编码")
    @TableField("campus_code")
    private String campusCode;

    @Schema(description = "楼层编码")
    @TableField("floor_code")
    private String floorCode;

    @Schema(description = "房间ID")
    @TableField("room_id")
    private Long roomId;

    @Schema(description = "房间编码")
    @TableField("room_code")
    private String roomCode;

    @Schema(description = "床位ID")
    @TableField("bed_id")
    private Long bedId;

    @Schema(description = "床位编码")
    @TableField("bed_code")
    private String bedCode;

    @Schema(description = "申请日期")
    @TableField("apply_date")
    private LocalDate applyDate;

    @Schema(description = "入住日期")
    @TableField("check_in_date")
    private LocalDate checkInDate;

    @Schema(description = "预计退宿日期（临时住宿）")
    @TableField("expected_check_out_date")
    private LocalDate expectedCheckOutDate;

    @Schema(description = "状态：1待审核 2已通过 3已拒绝 4已入住")
    @TableField("status")
    private Integer status;

    @Schema(description = "审批实例ID")
    @TableField("approval_instance_id")
    private Long approvalInstanceId;

    @Schema(description = "审核人ID")
    @TableField("approver_id")
    private Long approverId;

    @Schema(description = "审核人姓名")
    @TableField("approver_name")
    private String approverName;

    @Schema(description = "审核时间")
    @TableField("approve_time")
    private LocalDateTime approveTime;

    @Schema(description = "审核意见")
    @TableField("approve_opinion")
    private String approveOpinion;

    @Schema(description = "申请原因/备注")
    @TableField("apply_reason")
    private String applyReason;

    @Schema(description = "备注")
    @TableField("remark")
    private String remark;
}
